core/src/utils.ts

```diff
@@ -37,20 +37,35 @@ export function pad32Bytes(data) {
 }
 
 let slackSentTime = new Date().getTime()
-let errMsg = null
+let errMsg: string | null = null
 
-async function sendToSlack(error) {
+export async function sendToSlack(e: Error) {
   if (SLACK_WEBHOOK_URL) {
-    const e = error[0]
     const webhook = new IncomingWebhook(SLACK_WEBHOOK_URL)
+    let errorObj = {}
+    if (Array.isArray(e)) {
+      errorObj = {
+        message: e[0]?.message,
+        stack: e[0]?.stack,
+        name: e[0]?.name
+      }
+    } else {
+      errorObj = {
+        message: e.message,
+        stack: e.stack,
+        name: e.name
+      }
+    }
     const text = ` :fire: _An error has occurred at_ \`${os.hostname()}\`\n \`\`\`${JSON.stringify(
-      e
+      errorObj
     )} \`\`\`\n>*System information*\n>*memory*: ${os.freemem()}/${os.totalmem()}\n>*machine*: ${os.machine()}\n>*platform*: ${os.platform()}\n>*upTime*: ${os.uptime()}\n>*version*: ${os.version()}
    `
 
     try {
-      if (e && e.message && errMsg === e.message) {
+      // if the same error message is sent to slack before
+      if (e?.message && errMsg === e.message) {
         const now = new Date().getTime()
+        // if it's over 1 min since the last message was sent
```

Comment:
the logic there is to re-send the same message to slack after 1 min earliest. So we first check if the same error message has already been sent and if so has it been at least a minute since the last time it was sent. If all conditions met we resend the same error again